core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2323,31 +2323,75 @@ private[spark] object Utils
    * configure a new log4j level
    */
   def setLogLevel(l: Level): Unit = {
-    val (ctx, loggerConfig) = getLogContext
+    val (ctx, loggerConfig) = getLogContext()
     loggerConfig.setLevel(l)
     ctx.updateLoggers()
 
     // Setting threshold to null as rootLevel will define log level for spark-shell
     Logging.sparkShellThresholdLevel = null
   }
 
+  /**
+   * configure a new log4j level for specific package or class
+   *
+   * @param name package or class name such as "org.apache.spark" or
+   *             "org.apache.spark.SparkContext"
+   * @param level new level of [[org.apache.log4j.Level]]
+   */
+  def setLogLevel(name: String, level: Level): Unit = {
+    val (ctx, loggerConfig) = getLogContext(name)
+    if (loggerConfig != null) {
+      loggerConfig.setLevel(level)
+      logInfo(s"Logger ${loggerConfig.getName} level changed into $level")
+    } else {
+      val newLoggerConfig = new LoggerConfig(name, level, true)
+      ctx.getConfiguration.addLogger(name, newLoggerConfig)
+      logInfo(s"Added new logger $name = $level")
+    }
+    ctx.updateLoggers()
+  }
+
+  /**
+   * get logger level for specific package or class
+   *
+   * @param name  package or class name such as "org.apache.spark" or
+   *              "org.apache.spark.SparkContext"
+   */
+  def getLoggerLevel(name: String): Option[Level] = {
+    val (ctx, loggerConfig) = getLogContext(name)
+    Option(loggerConfig).map(c => c.getLevel)
+  }
+
+  /**
+   * remove logger for specific package or class
+   *
+   * @param name package or class name such as "org.apache.spark" or
+   *             "org.apache.spark.SparkContext"
+   */
+  def removeLogger(name: String): Unit = {
```

Comment:
Do you really need get/remove? not sure it's necessary

Reply:
Currently, get/remove in `Utils` only used by tests, but we might consider expose them in `SparkContext` in the future.

Reply:
They should be private to Spark then and just make sure they're annotated as for testing only